File: articles/ada-lovelace-programming-and-the-imagination-of-computation/julia/bernoulli_simple.jl
function bernoulli(n)
    a = [0//1 for _ in 0:n]
    for m in 0:n
        a[m + 1] = 1//(m + 1)
        for j in m:-1:1
            a[j] = j * (a[j] - a[j + 1])
        end
    end
    return a[1]
end

println("B_8=", bernoulli(8))

File: articles/ada-lovelace-programming-and-the-imagination-of-computation/haskell/BernoulliSimple.hs
module Main where

-- Lightweight placeholder example: a real Bernoulli implementation would use rationals carefully.
operationSequence :: [String]
operationSequence = ["initialize", "store", "multiply", "subtract", "repeat", "output"]

main :: IO ()
main = putStrLn ("operation_sequence=" ++ unwords operationSequence)
